crates/reporting/src/error/type.rs

```diff
@@ -513,6 +513,38 @@ pub fn type_problem<'b>(
                 severity,
             })
         }
+        TypeIsNotGeneralized(region, actual_type, generalizable) => {
+            let doc = alloc.stack([
+                alloc.reflow("This type variable has a single type:"),
+                alloc.region(lines.convert_region(region), severity),
+                if !generalizable.0 {
+                    alloc.concat([
```

Comment:
I notice that you didn't add a test case for this, and tried to come up with one myself. I don't know if this is possible to represent in current Roc. If you think it's possible to represent this in some future Roc (or in some way today that I don't know about), can we add a test case for it? Otherwise, maybe we should not be branching on generalization.